www/hujl00/sp2/js/main.js

```diff
@@ -0,0 +1,264 @@
+    const rightColumn = $('<div>').addClass('right-column');
+    const hints = $('<div>').addClass('hints');
+    const showHint2 = $('<button>').addClass('show-hint-button');
+    showHint2.text('Show second hint');
+
+    const shareButton = $('<button>').addClass('share-facebook');
+    shareButton.text('Share on Facebook');
+
+    let numberOfTrials = 8;
+    const trialsLeft = $('<div>').addClass('space-for-trials');
+    const updateTrialsColor = () => {
+        const color = numberOfTrials >= 6 ? 'darkgreen' :
+            numberOfTrials === 5 ? 'yellowgreen' :
+                numberOfTrials === 4 ? 'orange' :
+                    numberOfTrials <= 3 ? 'red' : 'black';
+
+        trialsLeft.css('color', color);
+    };
+    const spaceForWord = $('<div>').addClass('space-for-guessed-word');
+    const spaceForDefinition = $('<div>').addClass('space-for-definition');
+    const spaceForSynonyms = $('<div>').addClass('space-for-synonyms');
+    const end = $('<div>').addClass('end');
+
+    const keyboard = $('<div>').addClass('keyboard');
+    const keyboardFirstRow = $('<div>').addClass('keyboard-row');
+    const keyboardSecondRow = $('<div>').addClass('keyboard-row');
+    const keyboardThirdRow = $('<div>').addClass('keyboard-row');
+    const lettersFirstRow = ['Q', 'W', 'E', 'R', 'T', 'Y', 'U', 'I', 'O', 'P'];
+    const lettersSecondRow = ['A', 'S', 'D', 'F', 'G', 'H', 'J', 'K', 'L'];
+    const lettersThirdRow = ['Z', 'X', 'C', 'V', 'B', 'N', 'M'];
+    lettersFirstRow.forEach((letterK) => {
+        const key = $(`<div>${letterK}</div>`);
+        key.addClass('key');
+        keyboardFirstRow.append(key)
+    });
+    lettersSecondRow.forEach((letterK) => {
+        const key = $(`<div>${letterK}</div>`);
+        key.addClass('key');
+        keyboardSecondRow.append(key)
+    });
+    lettersThirdRow.forEach((letterK) => {
+        const key = $(`<div>${letterK}</div>`);
+        key.addClass('key');
+        keyboardThirdRow.append(key)
+    });
+    keyboard.append(keyboardFirstRow, keyboardSecondRow, keyboardThirdRow);
+
+    let points = parseInt(localStorage.getItem('hangmanPoints')) || 0;
+    const counters = $('<div>').addClass('count-points');
+    const countPoints = $('<div>').addClass('count-points');
+    countPoints.text('Total score: ' + points);
+
+    let gamesPlayed = parseInt(localStorage.getItem('hangmanGamesPlayed')) || 0;
+    const countGames = $('<div>').addClass('count-games');
+    countGames.text('Games played: ' + gamesPlayed);
+
+    const footer = $('<footer>').addClass('footer');
+    footer.text('Hangman © Lucie Hujerová 2024');
+
+    const spinner = $('<div>').addClass('spinner');
+    const showLoading = () => {
+        spaceForWord.append(spinner);
+    };
+    const hideLoading = () => {
+        spinner.remove();
+    };
+    const startGame = () => {
+        const getWordsURL = 'https://s3-us-west-2.amazonaws.com/s.cdpn.io/111863/dictionary.txt';

```

Comment:
tohle jsou 2MB statických dat, načítejte to předem při načtení stránky, ne až v momentě, kdy chcete začít hrát